src/serial/RS232Device.cc

```diff
@@ -37,6 +37,11 @@ bool RS232Device::getDCD(EmuTime::param /*time*/) const
 	return true; // TODO check
 }
 
+bool RS232Device::getRI(EmuTime::param /*time*/) const
+{
+	return true; // TODO check
+}
+
```

Comment:
That can be deduced from the schematics, all three I've found (Sony HB-G900, Sony HBI-232 and SVI-738) use the same line driver for the RS-232 input lines.
The Sony devices leave the inputs floating when there's no device plugged, resulting in a logical 0 being read from the inputs, the opposite of what it's implemented right now in OpenMSX. (The bit should be set to1 when reading the port)
The SVI on the other hand pulls them to 12v which reads as a logical 1, which is consistent with the current implementation (bit is set to 0 when reading the port).

Reply:
More on this, I tested on the SVI738 I have on loan.
inp(&h82) returns 0x7E
That matches my previous comment, the bits for DCD and CTS both are set to 0.
Also noticed something that I had missed before, the I8254 counter outputs are active low, but is implemented as active high right now.
Open data lines are set to 1.
On openMSX inp(&h82) with nothing plugged returns 0.

I propose invert the logic in readStatus(): start with `result = 0xFF;` and then mask out the active bits.

Reply:
Good idea (start from 0xff, and then clear specific bits).

It would also be great if you could add some comments about these experiments/results in the code. So that next year we still know which parts of the implementation are best guesses and which parts are verified on real MSX machines.

Thanks.

Reply:
Got tests back for the HX-22:
With the RS-232 switch in the back set to ON:
0xFF

Set to OFF:
0xBF

I'll write my full findings as of today here, will include a shorter version in the code.

Looking at this msx.org thread https://www.msx.org/forum/msx-talk/hardware/toshiba-hx-22?page=3, this machine doesn't have the RI pin but the inputs are left floating like on the Sony devices.

From the PCB pictures on msx.org:
ASCII MSXSerial-232: RI present, floating input lines
Sanyo MRS-001: No RI, floating input lines
Toshiba HX-R700: No RI, floating input lines

From the service manual:
SVI-737 Modem/RS-232 interface: RI present, pull-down lines (read the same as if floating on the port)

From the user manual:
SVI-757 RS-232 interface: RI present. This device is probably the same as the 737 without the modem parts populated, there's a good chance the input lines are also pull-down.